docker/synapse.Dockerfile

```diff
@@ -67,10 +67,11 @@ RUN /venv/bin/pip install -q --no-cache-dir \
 # Poetry runs multiple pip operations in parallel. Unfortunately this results
 # in race conditions when a dependency is installed while another dependency
 # is being up/downgraded in `scripts/synapse_sytest.sh`. Configure poetry to
-# serialize `pip` operations.
+# serialize `pip` operations by setting `experimental.new-installer` to a falsy
+# value.
 # See https://github.com/matrix-org/synapse/issues/12419
 # TODO: Once poetry 1.2.0 has been released, use the `installer.max-workers`
 #       config option instead.
-RUN poetry config experimental.new-installer false
+ENV POETRY_EXPERIMENTAL_NEW_INSTALLER ""
```

Comment:
poetry has a bug where this environment variable is not converted to a boolean, so we choose a falsy string value for it.
It's only fixed in poetry 1.2.0: https://github.com/python-poetry/poetry/commit/d4c2fe44f43d426ecf963bbce9325c45dbc40fc8

Reply:
Great find---can we stick this comment in the dockerfile though?